prow/caching.md

```diff
@@ -0,0 +1,50 @@
+# Caching dependencies in Prow
+
+Caching dependencies in prow works only when the [Pod Utilities][0] are enabled. A
+valid GCS Bucket has to be configured via `gcs_configuration` either in the Prow
+jobs `decoration_config` or in [Planks][1] `default_decoration_configs`.
+
+[0]: /prow/pod-utilities.md
+[1]: /prow/cmd/plank/README.md
+
+Saving and restoring the cache will be done in the `clonerefs` and `sidecar` pod
+utility containers, where:
+
+- `clonerefs` tries to restore the cache after the clone has been done. Multiple
+  `keys` can be specified to reference files inside the repository. Those files
+  will be hashed and serve as unique identifier for the cache.
+
+- `sidecar` tries to save the cache after the main test container has been
+  finished successfully. The specified `keys` will be used here as well to
+  calculate the target directory.
+
+We store the cache in Prows global root directory of the Bucket (`/prow/cache`),
+because the cache should be available between multiple Prow runs and even
+between different Prow jobs.
+
+A single cache entry can be defined in the `decoration_config` via:
+
+```yaml
+decoration_config:
+  caches:
+    - name: my-cache
+      path: /gocache
```

Comment:
So `path` is both the target (in case of download) and the source (in case of upload)?

Reply:
Could we get rid of the `name` attribute by using a sanitzed version of the `path` as name?

Reply:
Yes, I'll change the implementation to use the `Path` as indirect name. :+1: